src/screen.rs

```diff
@@ -1282,6 +1287,154 @@ impl Screen {
         }
         self.render();
     }
+    pub fn move_focus_left(&mut self) {
+        if self.terminals.is_empty() {
+            return;
+        }
+        if self.fullscreen_is_active {
+            return;
+        }
+        let active_terminal = self.get_active_terminal();
+        match active_terminal {
+            Some(active) => {
+                let next_index = self
+                    .terminals
+                    .iter()
+                    .enumerate()
+                    .filter(|(_, (_, c))| {
+                        c.is_directly_left_of(&active) && c.horizontally_overlaps_with(&active)
+                    })
+                    .max_by_key(|(_, (_, c))| c.get_horizontal_overlap_with(&active))
+                    .map(|(_, (pid, _))| pid);
+                match next_index {
+                    Some(p) => {
+                        self.active_terminal = Some(*p);
+                        ()
+                    }
+                    None => {
+                        self.active_terminal = Some(active.pid);
+                        ()
+                    }
+                }
+            }
+            None => {
+                self.active_terminal = Some(active_terminal.unwrap().pid);
+                ()
+            }
+        }
```

Comment:
I think this `match` statements should be reduced to `if let`, as it improves readability and that's what `if let` statements are for. And also reduces a level of indent :)